src/lib/getPeerDependenciesFromRegistry.ts

```diff
@@ -59,19 +60,40 @@ async function getPeerDependenciesFromRegistry(packageMap: Index<Version>, optio
     bar.render()
   }
 
-  return Object.entries(packageMap).reduce(async (accumPromise, [pkg, version]) => {
-    const dep = await packageManager.getPeerDependencies!(pkg, version, { cwd: options.cwd })
+  const packageEntries = Object.entries(packageMap)
+  const accum: Index<Index<string>> = {}
+
+  /**
+   * Processes peer dependencies for a package and updates the accumulator
+   * @param pkg - The package name
+   * @param version - The package version
+   * @returns Promise that resolves when peer dependencies are processed
+   */
+  const processPeerDependencies = async ([pkg, version]: [string, Version]) => {
```

Comment:
`processPeerDependencies` currently relies on a mutation side effect to return its results. Using a typical return statement so that this is a pure function with no side effects would be preferable.